### PR TITLE
[CORE24-287] feat(search): search config

### PR DESCRIPTION
**Pull request by internetti**
Jira ticket: JIRA-CORE24-287

## What have you implemented?

added the ui config for the search

### CODE REVIEW

File: libs/prm-components/src/lib/config/config.types.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export type EntityUIConfig = {
     sections: Section[];
   };
   list: unknown;
-  search: unknown;
+  search: Partial<EntityType>;
```
Comment (internetti) [inline libs/prm-components/src/lib/config/config.types.ts:64]: can't think of anything else needed right now ...

File: libs/prm-components/src/lib/config/participant.ts [hunk 2]
```diff
@@ -419,5 +419,205 @@ export const participantConfig: EntityUIConfig = {
...
+      label: 'Mother Name',
+    },
+    {
+      path: ['sex'],
+      dataType: DataType.String,
+      component: Component.Select,
+      label: 'Sex',
+    },
+    {
+      path: ['birthdate'],
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/config/participant.ts:467]: Is `path` meant to be the attribute name in the Participant? If that's the case, I think this should be `dateOfBirth`

File: libs/prm-components/src/lib/config/participant.ts [hunk 3]
```diff
@@ -419,5 +419,205 @@ export const participantConfig: EntityUIConfig = {
...
+      label: 'Phone number',
+    },
+    {
+      path: ['contactDetails.emails'],
+      dataType: DataType.String,
+      component: Component.TextInput,
+      label: 'Emails',
+    },
+    {
+      path: ['address'],
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/config/participant.ts:509]: `residence`

File: libs/prm-components/src/lib/config/participant.ts [hunk 4]
```diff
@@ -419,5 +419,205 @@ export const participantConfig: EntityUIConfig = {
...
+      label: 'Is minor',
+    },
+    {
+      path: ['nationalities'],
+      dataType: DataType.String,
+      component: Component.Select,
+      label: 'Nationality',
+    },
+    {
+      path: ['identifications'],
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/config/participant.ts:491]: If this is just for the number, I'd call it `identificationNumber`. But not sure what's the impact of that, if this references some other structure

File: libs/prm-components/src/lib/config/participant.ts [hunk 5]
```diff
@@ -419,5 +419,205 @@ export const participantConfig: EntityUIConfig = {
...
+      dataType: DataType.String,
+      component: Component.TextInput,
+      label: 'Primary Identification Number',
+    },
+    {
+      path: ['contactDetails.phones'],
```
Comment (dfigueroa87) [inline libs/prm-components/src/lib/config/participant.ts:497]: Same here, I'd just call this `phones` (and the next one `emails`)
Also, maybe we should rather use singular form? I'm guessing we'll be searching by a single value (This is actually maybe rather a question for my own schema PR)

File: libs/prm-components/src/lib/config/participant.ts (hunk 2 above)
Comment (internetti) [reply libs/prm-components/src/lib/config/participant.ts:467]: yeah, I did mean to refer to the participant's attributes, guess I got distracted while checking

File: libs/prm-components/src/lib/config/participant.ts (hunk 5 above)
Comment (internetti) [reply libs/prm-components/src/lib/config/participant.ts]: my thought was that we're searching in all of the phone entries of each participant. different ways to interpret, I could work with both, I now called them the same as in the zod schema

**Comment by dfigueroa87** [comment]
Changes look good, but there's too many commits in this PR now. Maybe needs to be rebased to `main`?

**Review by dfigueroa87 — APPROVED** [review]

**Pull request merged**